Content.Server/Nutrition/EntitySystems/FoodSystem.cs

```diff
@@ -227,7 +230,19 @@ private void OnDoAfter(EntityUid uid, FoodComponent component, ConsumeDoAfterEve
                 _utensilSystem.TryBreak(utensil, args.User);
             }
 
-            if (component.UsesRemaining > 0)
+            if (TryComp<StackComponent>(uid, out var stack)) 
+            {
+              //Not deleting whole stack piece will make troubles with grinder
+              if (stack.Count > 1) 
+              {
+                if (!forceFeed)
+                  args.Repeat = true;
```

Comment:
probably safe to just change this to be args.Repeat = !forceFeed, put it above the if (trycomp and remove it from below usesremaining too